### ISSUE TITLE
Some videos does not switch TV to HDR mode

### ISSUE THREAD

**Issue by dhlavaty** [nova-video-player/aos-AVP#405]
TL;DR; Player does not switch my TV to HDR mode, when playing some (all downloaded from YouTube for example) videos.

Long version:

Some HDR videos play ok, and my TV is correctly switched to HDR mode. For example "Sony Bravia OLED" video downloaded from https://www.demolandia.net/4k-video-test/hdr10/page-1.html?highlight=3

But none HDR video downloaded from Youtube, does not switch my TV to HDR mode. But same video files played with MXPlayer or using TV's built-in com.tcl.videoplayer works flawlessly. 

# How to reproduce:

1. Use this video https://youtu.be/7rzF_itNzsM (or any other HDR video from Youtube).
2. Download it using youtube-dl:

```sh
youtube-dl -f 337 https://www.youtube.com/watch\?v\=7rzF_itNzsM    
```
3. Play video using Nova Video Player on AndroidTV. TV should be switched to HDR mode, but it is not. Using MXPlayer or TCL's com.tcl.videoplayer player, it works ok.

My setup:

AndroidTV (version 9) on TCL 55C815
Nova Video Player 5.15.13-20201203.2027

**Comment by AlexisTM** [comment]
In my case, it does not switch to HDR and it completely fail to decode the colours for Dolby Vision, identically to VLC. (TCL C715)

Content used: https://4kmedia.org/lg-earth-dolby-vision-uhd-4k-demo/

**Comment by dhlavaty** [comment]
I can confirm @AlexisTM

Nova Video Player does not switch TV to **DOLBY VISION** mode for **LG Earth Dolby Vision UHD 4K Demo**.

But when I use TV's built-in **com.tcl.videoplayer** it turns on both DOLBY VISION and DOLBY AUDIO:

![IMG_7728](https://user-images.githubusercontent.com/513138/102700978-92a57300-4252-11eb-9f57-c04f650afee8.jpg)

**Comment by phhusson** [comment]
If it's just Dolby Vision support, then it's a dupe of https://github.com/nova-video-player/aos-AVP/issues/383 ?

**Comment by dhlavaty** [comment]
My original report was just about HDR10 not working. I'm not sure, if Dolby Atmos is related or not

**Comment by Neverends4** [comment]
Have you tried switching to customized decoding mode? Press "Force software decode" in option several times brings up hidden options.
For my buggy TCL OEM TV, using "android media player" as decoder fixes HDR. Under this mode Nova has exactly the same decoding capability as TV system player.

**Comment by dhlavaty** [comment]
I didn't know about press-FORCE-SOFTWARE-DECODE-option-several-times feature. (Why the h**k is it hidden in the first place?)

Switching to Android-MediaPlayer, forces my TV to switch to HDR mode correctly. BUT most video files I have and tested, just freezes after first 8-9 seconds. It just stops playing - video is frozen, sound stops playing. I can only exit the player.

And some video files (for example **LG Earth Dolby Vision UHD 4K Demo** mentioned above) does not play at all. It just stuck on black screen with "Loading video... 1%" text forever.

**Comment by Neverends4** [comment]
> I didn't know about press-FORCE-SOFTWARE-DECODE-option-several-times feature. (Why the h**k is it hidden in the first place?)
> 
> Switching to Android-MediaPlayer, forces my TV to switch to HDR mode correctly. BUT most video files I have and tested, just freezes after first 8-9 seconds. It just stops playing - video is frozen, sound stops playing. I can only exit the player.
> 
> And some video files (for example **LG Earth Dolby Vision UHD 4K Demo** mentioned above) does not play at all. It just stuck on black screen with "Loading video... 1%" text forever.

You experienced the very reason why it is hidden--it could cause more trouble.

I'm no programmer but to me Nova uses different "video mode" compared to other apps, like MX player, which has more success rate in switching TV to HDR mode.

**Comment by courville** [comment]
DV issue should be solved on 6.0.24 experimental release.

**Comment by courville** [comment]
Let's follow up on #383 for the remaining problems.
